services/tables/src/main/java/com/linkedin/openhouse/tables/controller/TablesController.java

```diff
@@ -198,6 +199,38 @@ public ResponseEntity<Void> deleteTable(
         apiResponse.getResponseBody(), apiResponse.getHttpHeaders(), apiResponse.getHttpStatus());
   }
 
+  @Operation(
+      summary = "RENAME Table",
+      description = "Renames a table resource",
+      tags = {"Table"})
+  @ApiResponses(
+      value = {
+        @ApiResponse(responseCode = "204", description = "Table RENAME: NO_CONTENT"),
+        @ApiResponse(responseCode = "400", description = "Table RENAME: BAD_REQUEST"),
+        @ApiResponse(responseCode = "401", description = "Table RENAME: UNAUTHORIZED"),
+        @ApiResponse(responseCode = "403", description = "Table RENAME: FORBIDDEN"),
+        @ApiResponse(responseCode = "404", description = "Table RENAME: TBL_DB_NOT_FOUND"),
+        @ApiResponse(responseCode = "409", description = "Table RENAME: TBL_EXISTS")
+      })
+  @PatchMapping(value = {"/v1/databases/{fromDatabaseId}/tables/{fromTableId}/rename"})
+  public ResponseEntity<Void> renameTable(
+      @Parameter(description = "Database ID", required = true) @PathVariable String fromDatabaseId,
+      @Parameter(description = "Table ID", required = true) @PathVariable String fromTableId,
+      @RequestParam(value = "toDatabaseId") String toDatabaseId,
+      @RequestParam(value = "toTableId") String toTableId) {
```

Comment:
I think we can make toTableId mandatory. Otherwise, supplying empty or null does not make sense for rename. As we don't support rename across db, we could skip toDatabaseId, but could also make it mandatory for consistency.

Reply:
I'll keep it mandatory for consistency, otherwise it would break on some of the comparison checks anyways

Reply:
By being RequestParams they are already forcefully required, I will add some unit tests around this.

| By default, method parameters that use this annotation are required, but you can specify that a method parameter is optional by setting the @RequestParam annotation’s required flag to false or by declaring the argument with a java.util.Optional wrapper.


https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-methods/requestparam.html